dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Voltage/TCL.mo

```diff
@@ -23,62 +23,74 @@ model TCL "Tap Changer Lock (TCL)"
 
   public
 
+    type State = enumeration (Standard "1: TCL is in normal state",
+                              Armed "2: TCL is armed",
+                              Locked "3: TCL is locked");
+
     parameter Types.VoltageModule UMin "Minimum voltage threshold before tap-changer locking";
     parameter Types.Time tLagBeforeLocked "Time to wait before activating lock event";
     parameter Types.Time tLagTransLockedT "Time to wait before sending lock event to high voltage transformers";
     parameter Types.Time tLagTransLockedD "Time to wait before sending lock event to low voltage transformers";
 
     Connectors.ImPin UMonitored (value (unit = "kV")) "Monitored voltage";
-    Connectors.BPin lockOrder (value (start = locked0)) "TCL lock order";
+    Connectors.BPin lockOrder (value (start = locked0)) "TCL manual lock order";
+    Connectors.BPin unlockOrder (value (start = locked0)) "TCL manual lock order";
     Boolean lockedT (start = locked0) "High voltage transformers locked ?";
     Boolean lockedD (start = locked0) "Low voltage transformers locked ?";
 
   protected
 
     parameter Boolean locked0 = false "Is the TCL initially locked ?";
+    parameter State state0 = State.Standard "Initial state";
 
     Boolean UUnderMin (start = false) "U < Umin ?";
     Types.Time tUnderUmin (start = Constants.inf) "Time when U < Umin";
     Types.Time tLocked (start = Constants.inf) "Time when the TCL was locked";
-    Boolean locked (start = locked0) "TCL locked ?";
+    State state(start = state0) "State of the automata throughout the simulation";
 
   equation
 
     // Check when the monitored voltage goes below UMin
-    // If the TCL is manually locked we ignore the voltage related events. Might be reviewed later.
-    when UMonitored.value < UMin and UMonitored.value > 0  and not (lockOrder.value) then
+    when UMonitored.value < UMin then
       UUnderMin = true;
       tUnderUmin = time;
-      Timeline.logEvent1(TimelineKeys.TapChangersArming);
-    elsewhen UMonitored.value > UMin and pre(UUnderMin) and not pre(lockOrder.value) then
+    elsewhen UMonitored.value >= UMin and pre(UUnderMin) then
       UUnderMin = false;
       tUnderUmin = Constants.inf;
-      Timeline.logEvent1(TimelineKeys.TapChangersUnarming);
     end when;
 
-    // Lock order activation
-    when ( (UUnderMin and time - tUnderUmin >= tLagBeforeLocked) or lockOrder.value ) and (not pre(locked)) then
-      locked = true;
+    // Automata state transitions
+    // Note that the deactivation from the Locked state is only manual
+    when UUnderMin and pre(state) == State. Standard then
```

Comment:
I think it's well handled by OM but in any case, it would be better to have State.Standard (no blank in between)